### ISSUE TITLE
automerge not working

### ISSUE THREAD

**Issue by mbrevda** [renovatebot/config-help#82]
Based on my config, it would see that all dev dependencies should be automerged, yet Im receiving a PR for them instead. Here is the config:

```json
{
  "extends": ["config:base"],
  "rebaseStalePrs": true,
  "devDependencies": {
    "automerge": true
  },
  "automergeType": "branch-push",
  "automerge": true,
  "major": {
    "automerge": false
  }
}

```

**Comment by rarkins** [comment]
Is it a public repository and you have a link to an example PR?

**Comment by mbrevda** [comment]
Its private. What info should I provide?

**Comment by rarkins** [comment]
Find the log file for exactly when the PR was created, search for all references to “automerge” for that branch.  Maybe you have a branch protection on master that is blocking it, for instance.

**Comment by rarkins** [comment]
Yes it does, because it’s evaluated after. What’s your requirement in plain English? Eg automerge everything except major updates to “dependencies”?

**Comment by mbrevda** [comment]
yup

**Comment by rarkins** [comment]
Try replacing your config with this:

```json
{
  "extends": ["config:base"],
  "rebaseStalePrs": true,
  "automergeType": "branch-push",
  "automerge": true,
  "packageRules": [{
	"depTypeList": ["dependencies"],
	"updateTypes": ["major"],
    "automerge": false
  }]
}
```

**Comment by mbrevda** [comment]
Perfect, thanks!
